strawberryfields/io/xir_io.py

```diff
@@ -103,11 +130,27 @@ def from_xir(xir_prog):
 
 
 # pylint: disable=too-many-branches
-def from_xir_to_tdm(xir_prog):
-    prog = TDMProgram(xir_prog.options["N"], name=xir_prog.options.get("name", "xir"))
+def from_xir_to_tdm(xir_prog: xir.Program) -> TDMProgram:
+    """Convert an XIR Program to a ``TDMProgram``.
+
+    Args:
+        xir_prog (xir.Program): the input XIR program object
+
+    Returns:
+        Program: corresponding ``TDMProgram``
```

Comment:
Is this type slightly off?
```suggestion
        TDMProgram: corresponding ``TDMProgram``
```